### ISSUE TITLE
[FR] Clear queue

### ISSUE THREAD

**Issue by Tommartinz** [FoedusProgramme/Gramophone#195]
It should be great to have the option to clear the current queue that it's playing.

It could be implemented by a button in the queue page or by a swipe-down gesture in the mini player.

Right now the only way to clear the queue is by closing the app and removing it from the recent tab, but it should be useful to have an in-app solution.

**Comment by nift4** [comment]
you can just pause the player and it will exit after two minutes

**Comment by Tommartinz** [comment]
But it could be a great quality life feature to have the option of doing it whenever you want

**Comment by Tommartinz** [comment]
Some people doesn't know this behavior, and it could be more intuitive having it

**Comment by nift4** [comment]
Your goal is to hide Gramophone from currently active music players?

**Comment by Tommartinz** [comment]
No, it's to have a dedicated option to clear the queue that it's playing, without having to wait or to close the entire app

**Comment by nift4** [comment]
@Tommartinz In that case I'm confused.
Closing the app like you said does not clear the queue, nor the thing I told you. These do just terminate the app.

**Comment by Tommartinz** [comment]
In my case it does, but if in your case it doesn't do it, then it's important to have the option to clear the queue.
By the way, you know what I mean when I refer to clearing the queue.

**Comment by nift4** [comment]
Can you show me a video/screen recording? If it clears the queue it's a semi-serious bug.

**Comment by Tommartinz** [comment]
Oh I just did it and it didn't clear the queue, well, that's it, we need an option to clear the queue

**Comment by nift4** [comment]
Just curious, why do you want to clear the queue?

**Comment by Tommartinz** [comment]
Sometimes I just want to get rid of the current list and play specific songs not related to each other, and it's more comfortable for me clearing all and starting the queue again from scratch

**Comment by nift4** [comment]
@Tommartinz you can just click any song instead of using "play next", it replaces the entire queue

**Comment by Tommartinz** [comment]
You are right, I don't know, maybe as a quality life feature, this could be

**Comment by nift4** [comment]
added to backlog

**Comment by jfmcbrayer** [comment]
> @Tommartinz you can just click any song instead of using "play next", it replaces the entire queue

For me, there's exactly one situation where I really want to clear the queue rather than just make a selection and overwrite the current queue. That's when I want to have only one song in the queue, not that song and everything after it in the view I selected it from. Clear, then enqueue one by one. Often that's because I want it to stop playing after my selected song(s).

I suppose the workaround is to enqueue my selections at the end, then skip ahead to them. Same number of steps, but clearing seems cleaner.
